packages/arco-lib/src/components/Table/Table.tsx

```diff
@@ -231,13 +242,39 @@ export const Table = implementRuntimeComponent({
   }, [data, sortRule]);
 
   const currentPageData = useMemo(() => {
+    let data = sortedData;
     if (enablePagination && useCustomPagination) {
       // If the `useCustomPagination` is true, then no pagination will be done and data over the pagesize will be sliced
       // Otherwise it will automatically paginate on the front end based on the current page
-      return sortedData?.slice(0, pageSize);
+      data = sortedData?.slice(0, pageSize);
     }
-    return sortedData;
-  }, [pageSize, sortedData, enablePagination, useCustomPagination]);
+
+    // auto-generated row key
+    return rowKey === 'auto'
```

Comment:
Do we need to auto generate key for users? I think it rarely happens.

Reply:
If a default column is not unique as a rowKey, this can also lead to duplicate selections.

Reply:
Yes, but I think it should leave to users to decide which is the `rowKey` and make sure it is unique.
Users don't know the rule of auto generating `rowKey`, so even we generated it for user, they won't know what dost it look like.

Reply:
Removed